test/aws/osml/model_runner/api/test_image_request.py

```diff
@@ -2,53 +2,226 @@
 
 import unittest
 
-legacy_execution_role = "arn:aws:iam::012345678910:role/OversightMLBetaInvokeRole"
+import boto3
+from botocore.stub import Stubber
+
+from aws.osml.model_runner.api import InvalidS3ObjectException
+from aws.osml.model_runner.api.image_request import (
+    FeatureDistillationNMS,
+    ImageRequest,
+    ModelInvokeMode,
+    MRPostProcessing,
+    MRPostprocessingStep,
+)
+from aws.osml.model_runner.api.request_utils import validate_image_path
+from aws.osml.model_runner.app_config import BotoConfig
+from aws.osml.model_runner.common import FeatureDistillationAlgorithm
+from aws.osml.model_runner.sink import Sink, SinkFactory
 
 
 class TestImageRequest(unittest.TestCase):
     def setUp(self):
         """
-        Set up virtual DDB resources/tables for each test to use
+        Set up sample data for each test.
         """
         self.sample_request_data = self.build_request_data()
 
     def tearDown(self):
+        """
+        Clean up after each test.
+        """
         self.sample_request_data = None
 
     def test_invalid_data(self):
-        from aws.osml.model_runner.api.image_request import ImageRequest
+        """
+        Test ImageRequest with missing or invalid image_id.
+        """
+        ir = ImageRequest(self.sample_request_data, image_id="")
+        self.assertFalse(ir.is_valid(), "ImageRequest should be invalid without a valid image_id")
+
+    def test_invalid_job_id(self):
+        """
+        Test ImageRequest with missing job_id.
+        """
+        ir = ImageRequest(self.sample_request_data, job_id=None)
+        self.assertFalse(ir.is_valid())
```

Comment:
It looks like a lot of the test code reverted from pytest assertion syntax to unittest assertion syntax.  I suggest we continue to use the pytest syntax with python's native assert statement and adopt that as the convention throughout the project.  More info [here](https://laerciosantanna.medium.com/pytest-vs-unittest-navigating-pythons-testing-terrain-2569912a0286#:~:text=Assertions,messages%20and%20better%20error%20introspection.)

Reply:
Going to try and move to pytest asserts where applicable.

Reply:
will look at standardizing this to pytest where applicable.